homeassistant/components/omnilogic/__init__.py

```diff
@@ -0,0 +1,83 @@
+"""The Omnilogic integration."""
+import asyncio
+import logging
+
+from omnilogic import LoginException, OmniLogic, OmniLogicException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import COORDINATOR, DOMAIN, OMNI_API, POLL_INTERVAL
+from .omnilogic_common import OmniLogicUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Omnilogic component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Omnilogic from a config entry."""
+
+    conf = entry.data
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    polling_interval = conf[POLL_INTERVAL]
+    api = OmniLogic(username, password)
+
+    try:
+        await api.connect()
+        await api.get_telemetry_data()
+    except LoginException as error:
+        _LOGGER.debug("OmniLogic login error: %s", error)
+        raise ConfigEntryNotReady from error
```

Comment:
One thing I missed - is a `LoginException` raised when the credentials are invalid, or when there's an error in connecting while trying to login in? If it's the former, you should return False, rather than raising.

Reply:
OK have updated so LoginException returns False, OmniLogicException returns ConfigEntryNotReady to reflect this. Also added handling of OmniLogicException type to config_flow and tests for completeness.